### PR TITLE
🎉 Added get_all_features method to Sample class

### PR DESCRIPTION
**Pull request by tdaniel06**
Self-explanatory title, see Issue #386 for context

### CODE REVIEW

File: src/plaid/containers/sample.py [hunk 1]
```diff
@@ -263,6 +263,18 @@ def get_all_features_identifiers_by_type(
...
+        """Get the list of all CGNS paths for all fields and global scalars."""
+        flat_tree, _ = CGH.flatten_cgns_tree(self.features.get_tree())
+        fn = self.get_field_names()
+        sn = self.get_scalar_names()
+        all_features_paths = []
+        for path in flat_tree:
+            name = path.split("/")[-1]
+            if name in fn or name in sn:
+                all_features_paths.append(path)
+        return all_features_paths
```
Comment (casenave) [inline src/plaid/containers/sample.py]: I'd rather have a `get_all_features_by_type(self, type)` function, where we use the get_feature_details_from_path function to filter types.
Because a feature is more than a global or a field.
Besides, scalar in get_scalar_names is deprecated, only the global notion exists anymore.

**Review by casenave — CHANGES_REQUESTED** [review]
Thanks for contributing!

File: src/plaid/containers/sample.py (hunk 1 above)
Comment (tdaniel06) [reply src/plaid/containers/sample.py]: I am currently trying your approach. The problem with returning all paths from the flattened tree whose types correspond to an input 'type' argument is that we also retrieve incomplete paths that do not correspond to any feature.
For example (from unit tests):
- get_all_features_by_type('field') gives: ['Base_2_2/Zone/VertexFields', 'Base_2_2/Zone/VertexFields/GridLocation', 'Base_2_2/Zone/VertexFields/OriginalIds', 'Base_2_2/Zone/VertexFields/test_node_field_1', 'Base_2_2/Zone/VertexFields/big_node_field', 'Base_2_2/Zone/CellCenterFields', 'Base_2_2/Zone/CellCenterFields/GridLocation', 'Base_2_2/Zone/CellCenterFields/OriginalIds', 'Base_2_2/Zone/CellCenterFields/test_elem_field_1'];
- get_all_features_by_type('global') gives: ['Global', 'Global/Time', 'Global/Time/IterationValues', 'Global/Time/TimeValues', 'Global/r', 'Global/test_scalar_1']

Comment (tdaniel06) [reply src/plaid/containers/sample.py]: For "Global", i can solve the issue directly with the 'sub_type' information returned by get_feature_details_from_path. For fields there is no equivalent to sub_type. I think i'll return 'other' type for non-leaf paths instead of 'field'

Comment (tdaniel06) [reply src/plaid/containers/sample.py]: I did a small fix, here are the results for different types:

- field: ['Base_2_2/Zone/VertexFields/GridLocation', 'Base_2_2/Zone/VertexFields/OriginalIds', 'Base_2_2/Zone/VertexFields/test_node_field_1', 'Base_2_2/Zone/VertexFields/big_node_field', 'Base_2_2/Zone/CellCenterFields/GridLocation', 'Base_2_2/Zone/CellCenterFields/OriginalIds', 'Base_2_2/Zone/CellCenterFields/test_elem_field_1']
- global: ['Global/r', 'Global/test_scalar_1']
- coordinate: ['Base_2_2/Zone/GridCoordinates/CoordinateX', 'Base_2_2/Zone/GridCoordinates/CoordinateY']
- boundary_condition: ['Base_2_2/Zone/ZoneBC/tag', 'Base_2_2/Zone/ZoneBC/tag/PointList', 'Base_2_2/Zone/ZoneBC/tag/GridLocation'] (can be further filtered via available sub_type)
- elements: ['Base_2_2/Zone/Elements_TRI_3/ElementRange', 'Base_2_2/Zone/Elements_TRI_3/ElementConnectivity']

File: tests/containers/test_sample.py [hunk 2]
```diff
@@ -1531,6 +1531,29 @@ def test_get_all_features_identifiers(self, sample_with_tree_and_scalar):
...
+
+        feat_paths = sample_with_tree_and_scalar.get_all_features_by_type('global')
+        assert "Global/r" in feat_paths
+        assert "Global/test_scalar_1" in feat_paths
+
+        feat_paths = sample_with_tree_and_scalar.get_all_features_by_type('coordinate')
+        assert 'Base_2_2/Zone/GridCoordinates/CoordinateX' in feat_paths
+        assert 'Base_2_2/Zone/GridCoordinates/CoordinateY' in feat_paths
+
+        feat_paths = sample_with_tree_and_scalar.get_all_features_by_type('boundary_condition')
```
[the remark this answers is not in the document]
Comment (tdaniel06) [reply tests/containers/test_sample.py]: @casenave no assert for now regarding boundary condition features in unit tests, I don't know which ones are relevant amont ['Base_2_2/Zone/ZoneBC/tag', 'Base_2_2/Zone/ZoneBC/tag/PointList', 'Base_2_2/Zone/ZoneBC/tag/GridLocation']

[the remark this answers is not in the document]
Comment (tdaniel06) [reply tests/containers/test_sample.py]: kept all of them and added assert in unit tests for type='boundary_condition'

**Review by casenave — APPROVED** [review]
thanks!
